pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -326,10 +328,50 @@ func (pl *VolumeRestrictions) EventsToRegister() []framework.ClusterEventWithHin
 		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add}},
 		// Pods may fail to schedule because the PVC it uses has not yet been created.
 		// This PVC is required to exist to check its access modes.
-		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add | framework.Update},
```

Comment:
Is it safe to remove the PVC's update event? The current Hint function will always ignore the event. After reading the plugin implementation, no update event registered makes sense to me.

I'd like to hear about others' opinions. @sanposhiho @AxeZhan 

Related to https://github.com/kubernetes/kubernetes/pull/122158

Reply:
The access modes can not be changed after a pvc is created.

Reply:
Unlike node events, the scheduler never ingore the creation event of a pvc. 

https://github.com/kubernetes/kubernetes/blob/6938c29a38e2b55b1a93f63af8c9a98aaeb5c026/pkg/scheduler/eventhandlers.go#L441

Reply:
I'm +1 for removing update event, since related changes to the PVC spec are blocked by immutability protection.

Not sure why we added it first, the comments said it's a future-proof, need opinions from reviewers of https://github.com/kubernetes/kubernetes/pull/103082

Reply:
/cc @alculquicondor @chrishenzie @ahg-g

Reply:
+1 to remove, but must be confirmed with someone from SIG Storage.

/hold

Reply:
/cc @jsafrane @xing-yang 

Can you take a look?

Reply:
https://github.com/kubernetes/kubernetes/pull/103082/files#r661392954

The question is whether anything has changed since that comment was made, whether any change to the PVC spec or status could cause a Pod to schedule.

ping @jsafrane @xing-yang

Reply:
Please remove Update, given, that, effectively, it doesn't matter.

Reply:
removed, after this PR merge we close https://github.com/kubernetes/kubernetes/pull/122158 as we no longer need it.